ksqldb-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -303,21 +303,25 @@ private void ensureKeySchemasMatch(
         .getSchemaTranslator(keyFormat.getFormatInfo().getProperties())
         .toParsedSchema(keySchema);
 
+    final Optional<SchemaMetadata> latest;
     try {
-      final SchemaMetadata latest = serviceContext.getSchemaRegistryClient()
-          .getLatestSchemaMetadata(
-              KsqlConstants.getSRSubject(dataSource.getKafkaTopicName(), true));
-      if (!latest.getSchema().equals(schema.canonicalString())) {
-        throw new KsqlException("Cannot INSERT VALUES into data source " + dataSource.getName()
-            + ". ksqlDB generated schema would overwrite existing key schema."
-            + "\n\tExisting Schema: " + latest.getSchema()
-            + "\n\tksqlDB Generated: " + schema.canonicalString());
-      }
-    } catch (IOException | RestClientException e) {
+      latest = SchemaRegistryUtil.getLatestSchema(
+          serviceContext.getSchemaRegistryClient(),
+          dataSource.getKafkaTopicName(),
+          true);
+
+    } catch (final Exception e) {
```

Comment:
Do we ever expect exceptions other than KsqlException here?

Reply:
No, we shouldn't have any other exception type here. I'll consolidate it